perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -518,6 +592,178 @@ def SingleStreamThroughputBenchmark(storage_schema, host_to_connect=None):
+  for thread in threads:
+    thread.join()
+  logging.info('All threads complete.')
+
+  try:
+    num_results = 0
+    while True:
+      res = result_queue.get_nowait()
+      num_results += 1
+      stream_num = res['stream_num']
+      for start_time, latency in zip(res['start_times'], res['latencies']):
+        json.dump({'kind': res['kind'],
+                   'start_time': start_time,
+                   'latency': latency,
+                   'size': FLAGS.object_size_kb,
+                   'stream_num': stream_num},
+                  sys.stdout)
+        sys.stdout.write('\n')
+  except Queue.Empty:
+    pass
+
+  if num_results != FLAGS.num_streams:
+    raise Exception('Couldn\'t get results from all streams!')
+
+
+def MultiStreamWrites(storage_schema, host_to_connect=None):
+  """Run multi-stream write threads."""
+  RunMultiThreadedWorkersWriteResults(
+      WriteWorker,
+      (storage_schema,
+       host_to_connect,
+       FLAGS.object_size_kb,
+       FLAGS.objects_per_stream,
+       FLAGS.start_time),
+      storage_schema,
+      host_to_connect=host_to_connect)
+
+
+def MultiStreamReads(storage_schema, host_to_connect=None):
+  """Run multi-stream read threads."""
+  RunMultiThreadedWorkersWriteResults(
+      ReadWorker,
+      (storage_schema,
+       host_to_connect,
+       FLAGS.object_size_kb,
+       FLAGS.start_time),
+      storage_schema,
+      host_to_connect=host_to_connect)
+
+
+def SleepUntilTime(when, too_late=None):
+  """Sleep until a given time.
+
+  Args:
+    when: float. The time to sleep to, as a POSIX timestamp.
+
+    too_late: None or Exception. If when is before now + MIN_SLEEP,
+      raise the given exception. If too_late is None, log a message
+      but do not raise an exception.
+  """
+
+  now = time.time()
+  sleep_time = when - now
+  if sleep_time > MIN_SLEEP:
+    time.sleep(sleep_time)
+  elif too_late is not None:
+    raise too_late('Sleep time %s was too small' % sleep_time)
+  else:
+    logging.info('Sleep time %s was too small', sleep_time)
+
+
+OBJECTS_WRITTEN_FILE_PREFIX = 'pkb-objects-written-'
+
+
+def WriteWorker(storage_schema, host_to_connect,
+                object_size_kb, num_objects,
+                start_time, result_queue, worker_num):
+  payload_bytes, payload_string = GenerateWritePayload(
+      object_size_kb * BYTES_PER_KILOBYTE)
+
+  objects_written = []
+  start_times = []
+  latencies = []
+
+  if start_time is not None:
+    SleepUntilTime(start_time, too_late=Exception)
+
+  WriteObjectsFromBuffer(storage_schema, FLAGS.bucket,
+                         ('pkb_write_worker_%s' % worker_num),
+                         num_objects, payload_bytes, payload_string,
+                         object_size_kb * BYTES_PER_KILOBYTE,
+                         objects_written,
+                         object_start_times=start_times,
+                         latency_results=latencies,
+                         host_to_connect=host_to_connect)
+
```

Comment:
Add a comment here to detail:
1) why do we need to write the object names to a file.
2) what is the format of the file, and how the files are named.